src/allmydata/test/web/test_status.py

```diff
@@ -0,0 +1,152 @@
+# Tests for code in allmydata.web.status
+
+from bs4 import BeautifulSoup
+from twisted.web.template import flattenString
+from zope.interface import implementer
+
+from allmydata.interfaces import (
+    IDownloadResults,
+    IDownloadStatus,
+)
+from allmydata.web.status import DownloadStatusElement
+
+from .common import (
+    assert_soup_has_favicon,
+    assert_soup_has_text,
+    assert_soup_has_tag_with_content,
+)
+from ..common import TrialTestCase
+
+
+@implementer(IDownloadResults)
+class FakeDownloadResults(object):
+
+    def __init__(self,
+                 file_size=0,
+                 servers_used=None,
+                 server_problems=None,
+                 servermap=None,
+                 timings=None):
+        """
+        See IDownloadResults for parameters.
+        """
+        self.file_size = file_size
+        self.servers_used = servers_used
+        self.server_problems = server_problems
+        self.servermap = servermap
+        self.timings = timings
```

Comment:
Does it seem weird that there is a download results renderer but there is no actual download results object in Tahoe-LAFS to be rendered?

Reply:
Seems to be an oversight, and it probably happened because these are the less tended parts of Tahoe.

I have filed [3310](https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3310) ("implement `DownloadStatus.get_results()`")